ctapipe/io/tableloader.py

```diff
@@ -0,0 +1,308 @@
+            else:
+                table = read_table(self.h5file, TRIGGER_TABLE)
+
+        if self.load_dl2_geometry:
+
+            shower_geometry_group = self.h5file.root[GEOMETRY_GROUP]
+
+            for i, reconstructor in enumerate(shower_geometry_group._v_children):
+
+                geometry = read_table(self.h5file, f"{GEOMETRY_GROUP}/{reconstructor}")
+
+                # rename DL2 columns to explicit reconstructor
+                # TBD: we could skip this if only 1 reconstructor is present
+                # or simply find another way to deal with multiple reconstructions
+                for col in set(geometry.colnames) - {"obs_id", "event_id"}:
+                    geometry.rename_column(col, f"{reconstructor}_{col}")
+
+                if ((table is None) or (len(table) == 0)) and (i == 0):
+                    table = geometry
+                else:
+                    table = join(
+                        table, geometry, keys=["obs_id", "event_id"], join_type="left"
+                    )
+        return table
+
+    def read_telescope_events(self, tel_ids):
+
+        table = vstack([self.read_telescope_events_for_id(tel_id) for tel_id in tel_ids])
+
+        return table
+
+    def read_events(self, labels=None):
+        """Read telescope-based event information.
+
+        Parameters
+        ----------
+        labels: list
+            Any list combination of tel_ids, tel_types, or telescope_descriptions.
+
+        Returns
+        -------
+        table: astropy.io.Table
+            Table with primary columns "obs_id", "event_id" and "tel_id".
+        """
+
+        if labels is None:
+            tel_ids = self.subarray.tel.keys()
+        else:
+            tel_ids = get_tel_ids(self.subarray, labels)
+
+        if any([self.load_dl1_images, self.load_dl1_parameters, self.load_true_images]):
+            table = self.read_telescope_events(tel_ids)
+        else:
+            table = None
+
+        table = self.read_subarray_events(table)
+
+        return table
+
+    def read_events_by_tel_type(self, labels=None):
+        """Read telescope-based event information.
+
+        Parameters
+        ----------
+        labels: list
+            Any list combination of tel_ids, tel_types, or telescope_descriptions.
+
+        Returns
+        -------
+        tables: dict(astropy.io.Table)
+            Dictionary of tables organized by telescope types
+            with primary columns "obs_id", "event_id".
+        """
+
+        if labels is None:
+            tel_ids = self.subarray.tel.keys()
+        else:
+            tel_ids = get_tel_ids(self.subarray, labels)
+
+        selected_subarray = self.subarray.select_subarray(tel_ids)
+        selected_tel_types = selected_subarray.telescope_types
+
+        tables = {}
+        for tel_type in selected_tel_types:
+
+            tables[str(tel_type)] = self.read_events_for_type(tel_type)
+
+        return tables
+
+    def read_events_for_type(self, tel_type):
```

Comment:
Shuold this be public? It confused me what the difference between this and he `read_events_per_tel_type` was (until I tried it).  I think only the latter should be public, and this could be hidden  as private to simplify the API.

Reply:
I think it should, it's still valid to use only 1 tel_type, though if you do

read_event_by_tel_type(**1 tel type**)

you get the same thing

it's the same for me, we can make it private or not

Reply:
it was just to simplify the API, it's not a huge issue to keep it.  I just found that so many ways to read the data were confusing, and there is not an equivalent one for tel_id, but both are handled properly by the higher-level function, so maybe it's better just to expose that.

Reply:
are you referring to a function doing `by_tel_id`?
for tel_id I don't think it makes sense to have an analog one, as you can use `for_tel_id` to get 1 table for that specific tel_id, whereas it makes sense for tel_type because it keeps together the tel_ids of that tel_type

Reply:
I agree, my suggestion was just that if we hide this it is a simpler API, but I'm ok keeping it.